pkg/placement/what-resolver.go

```diff
@@ -324,9 +324,12 @@ func (wr *whatResolver) processObject(ctx context.Context, cluster logicalcluste
 		return true
 	}
 	rObj, err := rr.lister.Get(objName)
```

Comment:
Just curious, does this `Get` call
https://github.com/kcp-dev/kubernetes/blob/aaa28aa88b2dab8fdaa99deb945ad26d9b4baeb0/staging/src/k8s.io/client-go/tools/cache/listers.go#L150-L159?

Reply:
Possibly.

Reply:
Turns out it traces to https://github.com/kcp-dev/kubernetes/blob/aaa28aa88b2dab8fdaa99deb945ad26d9b4baeb0/staging/src/k8s.io/client-go/dynamic/dynamiclister/lister.go#L50-L59 but I don't know why a non-nil pointer could be returned with a non-nil error.

/lgtm

Reply:
I choose to not rely on implementation details that could change.
The real mystery is why there is a nil pointer fault on a value that is known to not equal `nil`.

/approve